### PR TITLE
feat: remove compatible mode

### PR DESCRIPTION
**Pull request by yqrashawn**
PORTAL-2378

1. remove preference options `useModernProviderAPI`, `overrideWindowDotConflux`
2. take over window.conflux by default
3. remove Developer Mode in Settings UI
4. remove PortalProvider

---
This change is [Reviewable](https://reviewable.io/reviews/conflux-chain/helios/819)

### CODE REVIEW

**Comment by angelia-yuqi-personal** [comment]
Task linked: [PORTAL-2378 去掉兼容 Portal 的功能，增加一个开关](https://app.clickup.com/t/3719212/PORTAL-2378)

File: packages/inpage/setup-inpage-provider.js [hunk 1]
```diff
@@ -20,42 +20,9 @@ function validateMessage(e) {
...
-    )
-      return
-    window.localStorage.setItem(FLUENT_OVERRIDE_WINDOW_DOT_CONFLUX, false)
-    window.location.reload()
-  }
-}
-
 function maybeOverrideWindowDotEthereum(overrideWindowDotEthereum) {
   if (overrideWindowDotEthereum) {
     if (!window.localStorage.getItem(FLUENT_OVERRIDE_WINDOW_DOT_ETHEREUM)) {
```
Comment (RickStormBringer) [inline packages/inpage/setup-inpage-provider.js:28]: Both key and value of localstorage are strings. Better use JSON.parse. (JSON.parse will throws a [SyntaxError] exception if the string to parse is not valid JSON. )
 such as  try { JSON.pars(...)} catch (e) {...}

File: packages/rpcs/wallet_setPreferences/index.js [hunk 2]
```diff
@@ -8,28 +8,34 @@ export const schemas = {
...
+}) => {
   const txRst = setPreferences(params)
 
-  if (!isUndefined(params?.useModernProviderAPI)) {
+  const {overrideWindowDotEthereum} = await wallet_getPreferences()
+
+  // if overrideWindowDotEthereum is set
+  if (!isUndefined(params?.overrideWindowDotEthereum)) {
     const apps = getApp()
     apps.forEach(app => {
```
Comment (RickStormBringer) [inline packages/rpcs/wallet_setPreferences/index.js:33]: The Dapp pages which does not connected with wallet ,will not receive the post message.
Maybe we should let users render page by themselves?
I have implemented a similar function for your reference (https://github.com/Conflux-Chain/helios/pull/806. Only need to see packages/rpcs/wallet_setPreferences/index.js and  packages/inpage/setup-inpage-provider.js)

File: packages/inpage/setup-inpage-provider.js (hunk 1 above)
Comment (yqrashawn) [reply packages/inpage/setup-inpage-provider.js:28]: There's no difference here since we only check the existance of the item. We never set it to false

Comment (RickStormBringer) [reply packages/inpage/setup-inpage-provider.js:28]: OK.That make sense

File: packages/rpcs/wallet_setPreferences/index.js (hunk 2 above)
Comment (yqrashawn) [reply packages/rpcs/wallet_setPreferences/index.js:33]: > he Dapp pages which does not connected with wallet ,will not receive the post message.
> Maybe we should let users render page by themselves?

I don't quite understand why let user manully render all pages when we are able to help them render the connected one.

Comment (RickStormBringer) [reply packages/rpcs/wallet_setPreferences/index.js:33]: The unconnected dapp does not get message. They still Call old wallet

File: packages/inpage/setup-inpage-provider.js (hunk 1 above)
Comment (yqrashawn) [reply packages/inpage/setup-inpage-provider.js:28]: I changed `true` to `'true'` here to emphasize the behavior

**Review by RickStormBringer — APPROVED** [review]

**Pull request merged**